src/internal/components/chart-popover/index.tsx

```diff
@@ -79,35 +79,53 @@ function ChartPopover(
   const baseProps = getBaseProps(restProps);
   const popoverObjectRef = useRef<HTMLDivElement | null>(null);
 
+  // In chart popovers, dismiss button is present when they are pinned, so both values are equivalent.
+  const isPinned = dismissButton;
+
   const popoverRef = useMergeRefs(popoverObjectRef, ref);
 
   useEffect(() => {
-    const onDocumentClick = (event: MouseEvent) => {
+    const handleClickOutside = (event: MouseEvent) => {
       if (
         event.target &&
-        !nodeBelongs(popoverObjectRef.current, event.target as Element) && // click not in popover
-        !nodeContains(container, event.target as Element) // click not in segment
+        !nodeBelongs(popoverObjectRef.current, event.target) && // click not in popover
+        !nodeContains(container, event.target) // click not in segment
       ) {
         onDismiss(true);
       }
     };
 
-    document.addEventListener('mousedown', onDocumentClick, { capture: true });
+    document.addEventListener('mousedown', handleClickOutside, { capture: true });
     return () => {
-      document.removeEventListener('mousedown', onDocumentClick, { capture: true });
+      document.removeEventListener('mousedown', handleClickOutside, { capture: true });
     };
   }, [container, onDismiss]);
 
-  // In chart popovers, dismiss button is present when they are pinned, so both values are equivalent.
-  const isPinned = dismissButton;
+  const handleMouseLeave = useCallback(
+    (event: React.MouseEvent) => {
+      if (onMouseLeave) {
+        onMouseLeave(event);
+      }
+
+      // Dismiss unpinned popover when mouse exits to its inner container or a diffrent chart
+      if (
```

Comment:
The current control of dismissing popover by mousemove is here https://github.com/cloudscape-design/components/blob/main/src/mixed-line-bar-chart/hooks/use-mouse-hover.ts#L136, could you move your code into the same place?